File: OJ/LeetCode/src/2022 - Convert 1D Array Into 2D Array [Simulation].cs
public class Solution
{
    public int[][] Construct2DArray(int[] original, int m, int n)
    {
        if (original.Length != m * n) return new int[][] { };

        int[][] res = new int[m][];
        for (int i = 0; i < m; i++)
        {
            int[] temp = new int[n];
            for (int j = 0; j < n; j++)
                temp[j] = original[i * n + j];
            res[i] = temp;
        }

        return res;
    }
}

File: OJ/LeetCode/src/2022 - Convert 1D Array Into 2D Array [Simulation].cpp
class Solution {
public:
    vector<vector<int>> construct2DArray(vector<int>& original, int m, int n) {
        if (original.size() != m * n) return {};

        vector<vector<int>> res;
        for (int i = 0; i < m; i++) {
            vector<int> temp;
            for (int j = 0; j < n; j++)
                temp.emplace_back(original[i * n + j]);
            res.emplace_back(temp);
        }

        return res;
    }
};